Add Python get started docs

usha-mandya opened the pull request:
Signed-off-by: Usha Mandya <usha.mandya@docker.com>

Adding getting started docs for Python

aiordache commented on language/python/develop.md:
```diff
@@ -4,8 +4,253 @@ keywords: python, local, development, run,
+
+In this module, we’ll walk through setting up a local development environment for the application we built in the previous modules. We’ll use Docker to build our images and Docker Compose to make everything a whole lot easier.
+
+In this module, we’ll walk through setting up a local development environment for the application we built in the previous modules. We’ll use Docker to build our images and Docker Compose to make everything a whole lot easier.
```
Duplicated paragraph

aiordache commented on language/python/develop.md:
```diff
@@ -4,8 +4,253 @@ keywords: python, local, development, run,
+
+In the above command, we used the same MySQL image to connect to the database but this time, we passed the ‘mysql’ command to the container with the `-h` flag containing the name of our MySQL container name. Press CTRL-D to exit the MySQL  interactive terminal.
+
+Okay, now that we have a running MySQL, let’s update `app.py` to use MySQL as a datastore. Let’s also add some routes to our server. One for fetching records and one for inserting records.
```
Should we add here a link to the app setup in the `Build images` page as a reminder? Just in case we land directly on this page or forgot our application setup :sweat_smile: 
I would find useful also to have the directory structure/tree for the python app as we have to edit the files next:
```
python-docker
|____ app.py
|____ requirements.txt
|____ Dockerfile
```

aiordache commented on language/python/develop.md:
````diff
@@ -4,8 +4,253 @@ keywords: python, local, development, run,
+```
+
+In the above command, we used the same MySQL image to connect to the database but this time, we passed the ‘mysql’ command to the container with the `-h` flag containing the name of our MySQL container name. Press CTRL-D to exit the MySQL  interactive terminal.
+
````
I would add a new section title here to make it clear we change focus. We discuss next how to connect our python application to the database running in a container.

aiordache commented on language/python/develop.md:
```diff
@@ -4,8 +4,253 @@ keywords: python, local, development, run,
+
+In this module, we’ll walk through setting up a local development environment for the application we built in the previous modules. We’ll use Docker to build our images and Docker Compose to make everything a whole lot easier.
+
+## Local database and containers
```
Should we change section title to `Running a database in a container/ containerized database` or something else to quickly describe what we talk about in this part? And maybe we can move the first paragraph in the intro part.

aiordache commented on language/python/develop.md:
````diff
@@ -4,8 +4,253 @@ keywords: python, local, development, run,
+First, let’s add the `mysql-connector-python `module to our application using pip.
+
+```shell
+$ pip install mysql-connector-python
````
Maybe it would be easier to manually add the dependency to `requirements.txt`. Same as in https://github.com/docker/awesome-compose/blob/master/nginx-flask-mysql/backend/requirements.txt

From what I understand `pip freeze` command dumps all installed local packages, not only the ones we need for the app. I may be wrong, I have never used `pip freeze` before :sweat_smile:

usha-mandya replied:
Flask==1.1.1
mysql-connector==2.2.9

aiordache commented on language/python/develop.md:
````diff
@@ -4,8 +4,253 @@ keywords: python, local, development, run,
+Now, let’s run our container.
+
+```shell
+$ docker run \
````
Maybe outline here that we add the container to the DB network.  This way we can access the DB by its container name. We can also add a comment in the python code where we connect to the DB.

usha-mandya commented:
@aiordache Addressed the review comments we discussed earlier:

- I've updated the section heading in develop.md so users can get a better understanding of the task they perform
- Created a new section in build-images.md on the Python app directory structure
- Added a link from the develop.md page to the directory structure and sample application to help users get a background of the updates in build-images.

PTAL.

StefanScherer commented on language/python/build-images.md:
````diff
@@ -4,8 +4,238 @@ keywords: python, build, images, dockerfile
+
+```shell
+$ cd /path/to/python-docker
+$ pip install Flask
````
I suggest to consistently use `pip3` instead of `pip`. We suggest to install Python 3.8 or later and I'm not sure about if the magic "pip3 = pip" on all Mac and Windows platforms. Maybe @aiordache can confirm?

StefanScherer commented on language/python/build-images.md:
````diff
@@ -4,8 +4,238 @@ keywords: python, build, images, dockerfile
+Let’s start our application and make sure it’s running properly. Open your terminal and navigate to the working directory you created.
+
+```shell
+$ python -m flask run
````
Same suggestion to use `python3` instead of `python`.
On my Mac I have eg. two version of Pythons installed, don't know when and why I have them, more because it was needed to run some tools. I'm not a Python developer, so I'm always confused 😅

StefanScherer commented on language/python/build-images.md:
````diff
@@ -4,8 +4,238 @@ keywords: python, build, images, dockerfile
+Once we have our `requirements.txt` file inside the image, we can use the `RUN` command to execute the command `pip3 install`. This works exactly the same as if we were running `pip3 install` locally on our machine, but this time the modules are installed into the image.
+
+```dockerfile
+RUN pip3 install -r requirements.txt
````
here we already have `pip3` 👍

usha-mandya commented:
Thanks @StefanScherer. That's right. We do have pip3 within some of the examples. I've updated pip to pip3 to be consistent.

StefanScherer commented on language/python/build-images.md:
````diff
@@ -4,8 +4,238 @@ keywords: python, build, images, dockerfile
+
+Switch back to the terminal where our server is running and you should see the following requests in the server logs. The data and timestamp will be different on your machine.
+
+```shel
````
nit: small typo

StefanScherer reviewed (approved):
LGTM 🐍
